core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -220,25 +219,19 @@ private ListenableFuture<InstanceData> startRefreshAttempt() {
       refreshRunning = true;
     }
 
-    // To avoid unreasonable SQL Admin API usage, use a rate limit to throttle our usage.
-    ListenableFuture<?> rateLimit =
-        executor.submit(
-            () -> {
-              logger.fine(
-                  String.format(
-                      "[%s] Refresh Operation: Acquiring rate limiter permit.", instanceName));
-              //noinspection UnstableApiUsage
-              forcedRenewRateLimiter.acquire();
-              logger.fine(
-                  String.format(
-                      "[%s] Refresh Operation: Acquired rate limiter permit. Starting refresh...",
-                      instanceName));
-            },
-            executor);
+    logger.fine(
+        String.format("[%s] Refresh Operation: Acquiring rate limiter permit.", instanceName));
+    ListenableFuture<Long> delay = rateLimitCalculator.acquireAsync(executor);
```

Comment:
... and replacing it with this because it uses executor.schedule() to wait without blocking an executor thread.